opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -303,45 +305,82 @@ func CountPVCsForNodePool(k8sClient k8s.K8sClient, cr *opsterv1.OpenSearchCluste
 
 // Delete a STS with cascade=orphan and wait until it is actually deleted from the kubernetes API
 func WaitForSTSDelete(k8sClient k8s.K8sClient, obj *appsv1.StatefulSet) error {
+	ctx := context.Background()
```

Comment:
I am afraid if this fresh ctx ignores reconcile timeout or shut.
how about using the same ctx passed from the root reconciler managed by controller runtime or creating a child context using context.WithTimeout or context.WithCancel?

Reply:
@josedev-union  Is this still a concern ?

Reply:
@prudhvigodithi i already made a commit to use parent context. So this is not a concern anymore.